### ISSUE TITLE
Issue upgrading to 22.03.2LTS on MSSQL

### ISSUE THREAD

**Issue by swicken-dotcms** [dotCMS/core#23036]
### Describe the bug
When upgrading to 22.03.2 there is an issue with an upgrade task on MSSQL. It produces the following error in the logs:

```
dotcms_22-dotcms-22lts-1 | 12:44:05.469 INFO db.DotDatabaseMetaData - Executing: ALTER TABLE analytic_summary_period DROP CONSTRAINT uq__analytic_summary__74444068
dotcms_22-dotcms-22lts-1 | 12:44:05.472 ERROR db.DotDatabaseMetaData - Error executing: ALTER TABLE analytic_summary_period DROP CONSTRAINT uq__analytic_summary__74444068
dotcms_22-dotcms-22lts-1 | 12:44:05.473 INFO runonce.Task220402UpdateDateTimezones - Unable to drop constraint `uq__analytic_summary__74444068`.
dotcms_22-dotcms-22lts-1 | 12:44:05.473 INFO runonce.Task220402UpdateDateTimezones - updating analytic_summary_period.full_date to datetimeoffset
dotcms_22-dotcms-22lts-1 | 12:44:05.476 ERROR db.HibernateUtil - ---------- DotHibernate: error on rollbackTransaction ---------------
dotcms_22-dotcms-22lts-1 | com.microsoft.sqlserver.jdbc.SQLServerException: Cannot invoke a rollback operation when the AutoCommit mode is set to "true".
dotcms_22-dotcms-22lts-1 | com.dotmarketing.exception.DotRuntimeException: An error occurred when updating table 'analytic_summary_period': The index 'UQ__analytic_summary__74444068' is dependent on column 'full_date'.
dotcms_22-dotcms-22lts-1 | at com.dotmarketing.startup.runonce.Task220402UpdateDateTimezones.executeUpgrade(Task220402UpdateDateTimezones.java:81)
dotcms_22-dotcms-22lts-1 | at com.dotmarketing.startup.StartupTasksExecutor.executeStartUpTasks(StartupTasksExecutor.java:133)
dotcms_22-dotcms-22lts-1 | at com.liferay.portal.servlet.MainServlet.init_aroundBody0(Ma
```

Related ticket: https://dotcms.zendesk.com/agent/tickets/108528

### To Reproduce
Please speak with me to obtain a backup of a database on 22.03.1

With that, simply restore the backup and run the upgrade to 22.03.2, it will fail with this message

### Expected behavior
The upgrade should complete successfully

### Acceptance criteria
- Upgrade should complete successfully
- The same changes applied to postgres databases should be applied to MSSQL databases

**Comment by fabrizzio-dotCMS** [comment]
Once I got access to a copy  of the db I was able to investigate the causes of the failing migration

A few things to notice here:

- The UT was failing because the ALTER TABLE was trying to drop a constraint when in reality the uniqueness is getting achieved via index.  in certain cases, the BD Admin might decide that indices are a better approach to sustain unique fields.  So in this case the fix is attempting another delete if we determine that the constraint name is for  "unique" with the prefix UQ_ 
- Once fixed the original problem the migration continued to fail this time due to unexpected  "default" value constraints present on tables `container_version_info, contentlet_version_info, link_version_info, template_version_info` They all have a default getDate associated to the field named `version_ts` on each one of them.  We need to verify if these changes 
were introduced by the customer or if they come from an older version of dotCMS.  because the present table structure does not have a default value on these `version_ts` fields.

If we determine the changes were introduced by the customer The PR will have to change to exclude dealing with their custom code.  Here's the PR https://github.com/dotCMS/core/pull/23051
